physionet-django/annotation/views.py

```diff
@@ -1,3 +1,86 @@
-from django.shortcuts import render
+from rest_framework import generics, permissions, status
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from django.shortcuts import get_object_or_404
+from django.http import JsonResponse
+import json
 
-# Create your views here.
+from annotation.models import Annotation, AnnotationCollection, AnnotationType
+from project.modelcomponents.publishedproject import PublishedProject
+from .serializers import (
+    AnnotationSerializer,
+    AnnotationCollectionSerializer,
+    AnnotationTypeSerializer,
+)
+from oauth2_provider.contrib.rest_framework import (
+    TokenHasReadWriteScope,
+    TokenHasScope,
+    OAuth2Authentication,
+)
+from oauth2_provider.decorators import protected_resource
+from oauth2_provider.views.generic import ProtectedResourceView
+
+from rest_framework.permissions import SAFE_METHODS
+
+
+class AnnotationsScope(TokenHasScope):
+    def get_scopes(self, request, view):
+        return (
+            ["annotations:view"]
+            if request.method in SAFE_METHODS
+            else ["annotations:edit"]
+        )
+
+
+class AnnotationCollectionCreateAPIView(generics.CreateAPIView):
+    """
+    POST: Create an AnnotationCollection
+    """
+
+    authentication_classes = [OAuth2Authentication]
+    permission_classes = [AnnotationsScope, IsAuthenticated]
+    serializer_class = AnnotationCollectionSerializer
+    queryset = AnnotationCollection.objects.all()
+    required_scopes = ["annotations:edit"]
+
+
+class AnnotationTypeCreateAPIView(generics.CreateAPIView):
+    """
+    POST: Create an AnnotationType
+    """
+
+    authentication_classes = [OAuth2Authentication]
+    permission_classes = [AnnotationsScope, IsAuthenticated]
+    serializer_class = AnnotationTypeSerializer
+    queryset = AnnotationType.objects.all()
+    required_scopes = ["annotations:edit"]
```

Comment:
`AnnotationsScope` overrides the `get_scopes` method, so I don't think the following line is doing anything:

`required_scopes = ["annotations:edit"]`